cognee/api/v1/agents/routers/get_agents_router.py

```
@@ -0,0 +1,110 @@
+    @router.post("/")
+    async def create_agent_endpoint(
+        name: str,
+        user: User = Depends(get_authenticated_user),
+    ) -> AgentWithApiKeyDTO:
+        try:
+            agent_user, api_key = await create_agent(name, user)
+        except UserAlreadyExists:
+            raise HTTPException(
+                status_code=409,
+                detail=f"Agent with name '{name}' already exists",
+            )
```

Ruff (0.15.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)